from_cpython/Objects/iterobject.c

```diff
@@ -75,7 +66,7 @@ calliter_next(calliterobject *it)
                 Py_CLEAR(it->it_sentinel);
             }
         }
-        else if (PyErr_ExceptionMatches(PyExc_StopIteration)) {
+        else if (PyErr_ExceptionMatches(PyExc_StopIteration) || PyErr_ExceptionMatches(PyExc_IndexError)) {
```

Comment:
I noticed there are some other parts of this pr that deal with IndexErrors (the seqiter stuff) which makes sense; maybe it was just a testing change?  it looks like test_iter still passes if this line is reverted.